database/models.go

```diff
@@ -27,6 +27,7 @@ type Category struct {
 	Description string
 	Color       string
 	UserID      uint
+	User        User
```

Comment:
We're starting to get into situations where every model embeds every other model, resulting in giant (picture the JSON) structs with dependencies all over the place.

Next PR will need to clean these up, we only reference the `ID` for the needed resource and our endpoints will have custom result types as listed in:
```
https://gorm.io/docs/query.html#Joins
```

`user` can have `categories`, sure, but the inverse is bonkers.